tests/core/providers/test_websocket_provider.py

```diff
@@ -64,6 +60,13 @@ def test_websocket_provider_timeout(w3):
         w3.eth.accounts
 
 
+@patch('web3.providers.websocket.WebsocketProvider.coro_make_request')
+def test_websocket_provider_future_timeout(coro_make_request_mock, w3):
+    coro_make_request_mock.side_effect = asyncio.coroutine(lambda o: asyncio.sleep(0.02))
```

Comment:
This needs to me mocked out or else the timeout will be thrown from within coro_make_request and will never reach `future.result()`